### PR TITLE
Add deployment tooling for Franklin

### PR DESCRIPTION
**Pull request by rbreslow**
## Overview

This PR utilizes the remote state outputs from Raster Foundry (ECS cluster, RDS instance) to add an ECS service/ALB for Franklin.

I initialized a new database within the Raster Foundry RDS instance out-of-band:

```bash
$ psql -U raster_foundry -h database.service.rasterfoundry.internal
Password for user raster_foundry:
psql (9.6.11)
SSL connection (protocol: TLSv1.2, cipher: ECDHE-RSA-AES256-GCM-SHA384, bits: 256, compression: off)
Type "help" for help.

raster_foundry=> create database franklin;
CREATE DATABASE
```

Resolves https://github.com/azavea/raster-foundry-platform/issues/982

## Testing Instructions

Follow the instructions in the README to execute migrations:
```
Mar 18 11:54:38 52.55.128.27 franklin-api-migrations:  [ioapp-compute-0] INFO org.flywaydb.core.internal.license.VersionPrinter - Flyway Community Edition 6.2.4 by Redgate
Mar 18 11:54:38 52.55.128.27 franklin-api-migrations:  [ioapp-compute-0] INFO org.flywaydb.core.internal.database.DatabaseFactory - Database: jdbc:postgresql://database.service.rasterfoundry.internal:5432/franklin (PostgreSQL 9.6)
Mar 18 11:54:38 52.55.128.27 franklin-api-migrations:  [ioapp-compute-0] INFO org.flywaydb.core.internal.command.DbValidate - Successfully validated 3 migrations (execution time 00:00.017s)
Mar 18 11:54:38 52.55.128.27 franklin-api-migrations:  [ioapp-compute-0] INFO org.flywaydb.core.internal.schemahistory.JdbcTableSchemaHistory - Creating Schema History table "public"."flyway_schema_history" ...
Mar 18 11:54:38 52.55.128.27 franklin-api-migrations:  [ioapp-compute-0] INFO org.flywaydb.core.internal.command.DbMigrate - Current version of schema "public": << Empty Schema >>
Mar 18 11:54:38 52.55.128.27 franklin-api-migrations:  [ioapp-compute-0] INFO org.flywaydb.core.internal.command.DbMigrate - Migrating schema "public" to version 1 - Add Collection Items
Mar 18 11:54:47 52.55.128.27 franklin-api-migrations:  [ioapp-compute-0] INFO org.flywaydb.core.internal.command.DbMigrate - Migrating schema "public" to version 2 - Add Collections
Mar 18 11:54:47 52.55.128.27 franklin-api-migrations:  [ioapp-compute-0] INFO org.flywaydb.core.internal.command.DbMigrate - Migrating schema "public" to version 3 - add parent collection
Mar 18 11:54:47 52.55.128.27 franklin-api-migrations:  [ioapp-compute-0] INFO org.flywaydb.core.internal.command.DbMigrate - Successfully applied 3 migrations to schema "public" (execution time 00:09.671s)
```

See that the application is serving requests:

```bash
$ http --headers https://franklin.rasterfoundry.com/open-api/spec.yaml
HTTP/1.1 200 OK
Connection: keep-alive
Content-Length: 6111
Content-Type: text/plain; charset=UTF-8
Date: Wed, 18 Mar 2020 16:33:37 GMT
```

### CODE REVIEW

**Review by notthatbreezy — APPROVED** [review]
Works well for me! Looks good.

File: terraform/api.tf [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+resource "aws_lb_target_group" "api" {
+  name = "tg${var.project}API"
+
+  health_check {
+    healthy_threshold   = 3
+    interval            = 30
+    matcher             = 200
+    protocol            = "HTTP"
+    timeout             = 3
+    path                = "/"
```
Comment (hectcastro) [inline terraform/api.tf]: I believe that there was talk about using the OpenAPI YAML as the health check endpoint. Should that be used here vs. `/`?

File: terraform/api.tf [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+      status_code = "HTTP_301"
+    }
+  }
+}
+
+resource "aws_lb_listener" "api" {
+  load_balancer_arn = aws_lb.api.id
+  port              = 443
+  protocol          = "HTTPS"
+  certificate_arn   = data.terraform_remote_state.core.outputs.ssl_certificate_arn
```
Comment (hectcastro) [inline terraform/api.tf:73]: It is nice that we can do this.

**Review by hectcastro — APPROVED** [review]
Looks good. It'll be interesting to see how we connect this changes in the project repository. 👍

File: terraform/api.tf (hunk 1 above)
Comment (rbreslow) [reply terraform/api.tf]: Yeah, we can use that. I used `/` and heard back about the other endpoint being available after the PR was up.

**Pull request merged**